addons/sale_project/models/project.py

```diff
@@ -901,6 +901,16 @@ def _inverse_partner_id(self):
             if task.sale_order_id and task.partner_id.commercial_partner_id not in consistent_partners:
                 task.sale_order_id = task.sale_line_id = False
 
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        consistent_partners = (
+            self.sale_order_id.partner_id
+            | self.sale_order_id.partner_invoice_id
+            | self.sale_order_id.partner_shipping_id
+            ).commercial_partner_id
+        if self.sale_order_id and self.partner_id.commercial_partner_id not in consistent_partners:
+            self.sale_line_id = False
```

Comment:
shouldn't this be like _inverse_partner_id ?

```suggestion
            task.sale_order_id = task.sale_line_id = False
```

Reply:
Hello @RaoufGhrissi, shouldn't we follow the precedent that we change only the field value in its own compute or onchange function to avoid triggerring additional calls due to intermediate changes. Am I wrong ?. But when I tested this out previously there we no additional triggers. If our aim is remove the inverses then I guess it wouldn't be much of a problem here to make `sale_order_id` also to be set `False`. What do you think ?

Reply:
I have pushed the changes with minute changes in test case. You can have a look.